src/core/lib.lua

```diff
@@ -386,12 +353,64 @@ if ffi.abi("be") then
    function htonl(b) return b end
    function htons(b) return b end
 else
-   function htonl(b) return bit.bswap(b) end
-   function htons(b) return bit.rshift(bit.bswap(b), 16) end
+   function htonl(b) return bswap(b) end
+   function htons(b) return rshift(bswap(b), 16) end
 end
 ntohl = htonl
 ntohs = htons
 
+-- The fact that BitOps return signed integers is irritating, to say
+-- the least.  One counter-intuitive consequence is that, for example,
+--
+--   bit.bswap(bit.bswap(0x80000000)) == 0x80000000
```

Comment:
I don't think that a single bit.tobit() justifies a wrapper function any more. My original method was cumbersome enough to make it seem useful.

I've actually known the interoperation argument before, but I haven't really understood why the cross-platform problem can't be solved without exposing a counter-intuitive API. Oh, well :)